main.tf

```diff
@@ -0,0 +1,117 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = ">= 3.8"
+    }
+  }
+}
+
+provider "azurerm" {
+  features {}
+
+  subscription_id = "d33b95c7-af3c-4247-9661-aa96d47fccc0"
+  client_id       = "0568731c-73ad-44fa-89b6-8123ee27d574"
+  client_secret   = var.arm_client_secret
+  tenant_id       = "7d6f97d6-d755-4c10-b763-409cc4b6638f"
+}
+
+data "azurerm_resource_group" "main" {
+  name = "OpenCohort21_MatthewHolmes_ProjectExercise"
+}
+
+resource "random_string" "resource_code" {
+  length  = 5
+  special = false
+  upper   = false
+}
+
+resource "azurerm_storage_account" "main" {
```

Comment:
We usually wouldn't manage this storage account directly in Terraform - it confuses the order of things since the state wants to be in here before Terraform can start, but then the state can't be set up until this account exists!

Typically we'd manage these manually or in a separate repo where we don't mind checking in the state directly

Reply:
I've removed this resource from my PR.